src/main/java/org/apache/sling/feature/cpconverter/index/IndexDefinitions.java

```diff
@@ -74,6 +76,11 @@ public void addNode(@NotNull String parentPath, @NotNull DocViewNode2 node) {
         if ( existing != null ) {
             currentChildren.remove(existing);
         }
+        if ( CollectionUtils.isEmpty(node.getProperties())
+                && node.getName()!= null
+                && !(node.getName().toString().contains(".txt")) ){
```

Comment:
This check doesn't look right, I see the failing test in `IndexDefinitionsEntryHandlerTest#handleIndexDefinitionWithStopwordsInAnalyzer , but we should find a more generic condition besides "*.txt".